src/Features/WcPayPromotion/Init.php

```diff
@@ -101,10 +101,25 @@ public static function should_register_pre_install_wc_pay_promoted_gateway() {
 		}
 		$wc_pay_spec = self::get_wc_pay_promotion_spec();
 
-		if ( ! $wc_pay_spec ) {
+		if ( ! $wc_pay_spec || ! isset( $wc_pay_spec->additional_info ) ) {
 			return false;
 		}
-		return true;
+
+		$anon_id        = isset( $_COOKIE['tk_ai'] ) ? sanitize_text_field( wp_unslash( $_COOKIE['tk_ai'] ) ) : null;
+		$allow_tracking = 'yes' === get_option( 'woocommerce_allow_tracking' );
+		$abtest         = new \WooCommerce\Admin\Experimental_Abtest(
+			$anon_id,
+			'woocommerce',
+			$allow_tracking
+		);
+
+		$variation_name = $abtest->get_variation( 'woocommerce_wc_pay_promotion_payment_methods_table_' . $wc_pay_spec->additional_info->experiment_version );
```

Comment:
Do we need to double check that `experiment_version` exists above first or are we sure it will always exist on this API going forward?

This will be a public API, so anything we add here will continue to be used for the same request in the future with this version of WooCommerce Admin.

Reply:
Yeah I should probably do that as well